packages/vite/src/node/cli.ts

```diff
@@ -48,6 +53,16 @@ function cleanOptions<Options extends GlobalCLIOptions>(
   delete ret.filter
   delete ret.m
   delete ret.mode
+  for (const [key, value] of Object.entries(ret)) {
+    if (Array.isArray(value)) {
+      // may need to define the final return of option by themselves,
+      // so they retain their implementation portal.
+      // If it is not customized, take the last item of InlineConfig as the final result.
+      ret[key as keyof Options] = normalizeOptionsConfig[key]
+        ? normalizeOptionsConfig[key](value)
+        : value[value.length - 1]
+    }
+  }
```

Comment:
The logic here may be extracted and then applied to other options as well.

Reply:
OK, I'll try to modify it.